core/ustring.cpp

```diff
@@ -355,24 +357,24 @@ String &String::operator+=(const char *p_str) {
 		return *this;
 	}
 
-	int src_len = 0;
+	int rhs_len = 0;
 	const char *ptr = p_str;
 	while (*(ptr++) != 0) {
-		src_len++;
+		rhs_len++;
 	}
 
-	int from = length();
-
-	resize(from + src_len + 1);
+	const int lhs_len = length();
 
-	CharType *dst = ptrw();
+	resize(lhs_len + rhs_len + 1);
 
-	set(length(), 0);
+	CharType *dst = ptrw() + lhs_len;
 
-	for (int i = 0; i < src_len; i++) {
-		dst[from + i] = p_str[i];
+	for (int i = 0; i < rhs_len; i++) {
+		dst[i] = p_str[i];
 	}
 
+	dst[rhs_len] = 0;
```

Comment:
Can this be turned into a `memcpy()` too? And I think you can include the trailing null in it since `p_str` is null terminated.

Reply:
This can't be turned into `memcpy`, because characters in String are CharType, not char. `memcpy` won't handle it properly if CharType is not defined as char. But I think you are right about nullptr string terminator. I will change and test it later.